### PR TITLE
[touchwand] - Add support for Risco alarm system sensors and AcWand thermostat

### PR DESCRIPTION
**Pull request by roieg**
Add support for risco alarmsystem sensors and AcWand thermostat.

Improvement    - now support tow additional device types
* Risco alarm system sensors can be used as smoke , motion and door/window sensors 
* [AcWand](http://www.touchwand.com/products/touchwand-acwand/) thermostat fully supported ,able to control and read thermostat status , target room temperature , mode and fan level

### CODE REVIEW

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link TouchWandBSensorHandler} is responsible for handling command for Binary Sensor unit
+ *
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:31]:
```suggestion
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+ *
+ * @author Roie Geron - Initial contribution
+ *
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:34]:
```suggestion
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @Override
+    void touchWandUnitHandleCommand(Command command) {
+    }
+
+    void updateChannelDoorWindow(TouchWandBSensorUnitData unitData) {
+        OpenClosedType myOpenClose;
+        String isOpen = unitData.getCurrStatus();
+        logger.info("recieved status {} from door unit {} ", isOpen, unitData.getName());
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:77]:
```suggestion
        logger.debug("received status {} from door unit {} ", isOpen, unitData.getName());
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        } else {
+            logger.debug("TouchWandBSensorUnitData illegal update value {}", isOpen);
+            return;
+        }
+        updateState(CHANNEL_DOORWINDOW, myOpenClose);
+    }
+
+    void updateChannelMotion(TouchWandBSensorUnitData unitData) {
+        String motion = unitData.getCurrStatus();
+        logger.info("recieved status {} from motion unit {} ", motion, unitData.getName());
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:91]:
```suggestion
        logger.debug("received status {} from motion unit {} ", motion, unitData.getName());
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link TouchWandAlarmSensorHandler} is responsible for handling command for Alarm Sensor unit
+ *
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java:31]:
```suggestion
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+ *
+ * @author Roie Geron - Initial contribution
+ *
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandThermostatHandler.java:34]:
```suggestion
```

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandUnitDiscoveryService.java [hunk 7]
```diff
@@ -96,7 +95,7 @@ protected void startScan() {
                         }
                         String type = touchWandUnit.getType();
                         if (!Arrays.asList(SUPPORTED_TOUCHWAND_TYPES).contains(type)) {
-                            logger.debug("Unit discovery skipping unsupported unit type : {} ", type);
+                            logger.info("Unit discovery skipping unsupported unit type : {} ", type);
```
Comment (kaikreuzer) [inline bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/discovery/TouchWandUnitDiscoveryService.java:98]: Please keep debug here.

```suggestion
                            logger.debug("Unit discovery skipping unsupported unit type : {} ", type);
```

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
Thank you! Just some very small review comments.

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java (hunk 1 above)
Comment (roieg) [reply bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:31]: done

File: bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java (hunk 2 above)
Comment (roieg) [reply bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandBSensorHandler.java:34]: done

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**